docs/tsp/typeServerProtocol.ts

```diff
@@ -288,42 +315,33 @@ export namespace TypeServerProtocol {
         SettingsChange = 'typeServer/settingsChange',
```

Comment:
our typechecker settings are partially represented by VSCode settings (which I guess will have to be watched somehow in this protocol) and partially represented by config files. How will the type client understand when any arbitrary configuration changes? For this reason, I think it makes more sense to have a "push" based snapshot approach where the type server pushes these.

Reply:
Yeah this is a little backwards at the moment. The pylance extension tells VS code which configurations it is watching. But then it asks for the settings instead of being sent them. This should probably be changed to same pattern.

The Type server should query for settings, it shouldn't be told them.